tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/dav/DavClient.java

```diff
@@ -140,10 +143,10 @@ private Mono<Void> handleContactCreationResponse(HttpClientResponse response, St
         };
     }
 
-    public Mono<Void> updateCalendarObject(String username, URI calendarObjectUri, UnaryOperator<DavCalendarObject> calendarObjectUpdater) {
-        return getCalendarObjectByUri(username, calendarObjectUri)
+    public Mono<Void> updateCalendarObject(DavUser user, URI calendarObjectUri, UnaryOperator<DavCalendarObject> calendarObjectUpdater) {
+        return getCalendarObjectByUri(user, calendarObjectUri)
```

Comment:
It looks like, to me that we are fetching the object twice...

Or report only contains the URI and not the VEvents ?